dockerfile_parse/parser.py

```diff
@@ -667,12 +667,12 @@ def add_lines_at(self, anchor, *lines, **kwargs):
         replace = kwargs.pop('replace', False)
         after = kwargs.pop('after', False)
         assert not (after and replace)
-        assert not kwargs, "Unknown keyword argument(s): {0}".format(kwargs.keys())
+        assert not kwargs, "Unknown keyword argument(s): {0}".format(list(kwargs.keys()))
 
         # find the line number for the insertion
         df_lines = self.lines
         if isinstance(anchor, int):  # line number, just validate
-            assert anchor in range(len(df_lines))
+            assert anchor in list(range(len(df_lines)))
```

Comment:
Is there a better way to fix this warning? I'm pretty sure `X in range` is O(1) on python 3 but `X in list` is O(n). Maybe just silence it, or use a good old `0 <= x < ...`

Reply:
`X in range is O(1)` I'm not sure about this, it's still hidden iteration so I'd say O(n). But faster than alocating space for list, creating list and iterating list for sure.

Reply:
You could do something like
```python
def in_range(n, r):
    return r.start <= n < r.stop and ((n - r.start) % r.step) == 0
```
I'm sure if I thought of this, python people did too :D

Reply:
if you are sure, you can link cpython code here :D

Reply:
@chmeliik kudos
https://github.com/python/cpython/blob/master/Objects/rangeobject.c#L405

Reply:
@MartinBasti oh you actually found it, nice :D

Reply:
reducted kudos
it iterates sometimes: https://github.com/python/cpython/blob/master/Objects/rangeobject.c#L419